### ISSUE TITLE
Final allows redefinition if another final statement is used.

### ISSUE THREAD

**Issue by JulesGM** [python/mypy#13076]
mypy 0.961 (compiled: yes)

```python

a: Final[int] = 123
a: Final[int] = 321

```

doesn't cause an error.
I would have expected it to fail like 

```python

a: Final[int] = 123
a: int = 321

```

Is this done on purpose?

**Comment by AlexWaygood** [comment]
Can't reproduce: https://mypy-play.net/?mypy=latest&python=3.10&gist=2eb299e7b28965875f54af28475d5f64
